hs_composite_resource/models.py

```diff
@@ -0,0 +1,180 @@
+
+    class Meta:
+        verbose_name = 'Composite Resource'
+        proxy = True
+
+    @property
+    def can_be_public_or_discoverable(self):
+        # resource level metadata check
+        if not super(CompositeResource, self).can_be_public_or_discoverable:
+            return False
+
+        # filetype level metadata check
+        for lf in self.logical_files:
+            if not lf.metadata.has_all_required_elements():
+                return False
+
+        return True
+
+    def set_default_logical_file(self):
+        """sets an instance of GenericLogicalFile to any resource file objects of this instance
+        of the resource that is not already associated with a logical file. """
+
+        for res_file in self.files.all():
+            if not res_file.has_logical_file:
+                logical_file = GenericLogicalFile.create()
+                res_file.logical_file_content_object = logical_file
+                res_file.save()
+
+    @property
+    def supports_logical_file(self):
+        """ if this resource allows associating resource file objects with logical file"""
+        return True
+
+    def get_metadata_xml(self, pretty_print=True):
+        from lxml import etree
+
+        # get resource level core metadata as xml string
+        xml_string = super(CompositeResource, self).get_metadata_xml(pretty_print=False)
+        # add file type metadata xml
+
+        # create an etree xml object
+        RDF_ROOT = etree.fromstring(xml_string)
+
+        # get root 'Description' element that contains all other elements
+        container = RDF_ROOT.find('rdf:Description', namespaces=self.metadata.NAMESPACES)
+
+        for lf in self.logical_files:
+            lf.metadata.add_to_xml_container(container)
+
+        return etree.tostring(RDF_ROOT, pretty_print=pretty_print)
+
+    def supports_folder_creation(self, folder_full_path):
+        """this checks if it is allowed to create a folder at the specified path"""
+
+        path_parts = folder_full_path.split("/")
+        # remove the new folder name from the path
+        path_parts = path_parts[:-1]
+        path_to_check = "/".join(path_parts)
+        if not path_to_check.endswith("/data/contents"):
+            if self.resource_federation_path:
+                res_file_objs = self.files.filter(
+                    object_id=self.id,
+                    fed_resource_file_name_or_path__contains=path_to_check).all()
+            else:
+                res_file_objs = self.files.filter(object_id=self.id,
+                                                  resource_file__contains=path_to_check).all()
+            for res_file_obj in res_file_objs:
+                if not res_file_obj.logical_file.supports_resource_file_rename or \
+                        not res_file_obj.logical_file.supports_resource_file_move:
+                    return False
+
+        return True
+
+    def supports_move_or_rename_file_or_folder(self, src_full_path, tgt_full_path):
```

Comment:
We need to get away from these unwieldly names. Everything is logically a rename. 
Both files and folders are paths.